envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsLoggingModeTest.kt

```diff
@@ -0,0 +1,691 @@
+                  outgoing:
+                    dependencies:
+                      - service: "echo"
+                      - service: "echo2"
+        """.trimIndent()
+
+        private val echoEnvoy by lazy { envoyContainer1 }
+        private val echo2Envoy by lazy { envoyContainer2 }
+        private lateinit var echo3Envoy: EnvoyContainer
+
+        @Container
+        private val sourceIpClient = ToxiproxyContainer(exposedPortsCount = 2).withNetwork(network)
+        private val sourceIpClientToEchoProxy by lazy { sourceIpClient.createProxyToEnvoyIngress(envoy = echoEnvoy) }
+        private val sourceIpClientToEcho2Proxy by lazy { sourceIpClient.createProxyToEnvoyIngress(envoy = echo2Envoy) }
+
+        private val echoLocalService by lazy { localServiceContainer }
+        private val echo2LocalService by lazy { echoContainer2 }
+
+        @JvmStatic
+        @BeforeAll
+        fun setupTest() {
+            setup(appFactoryForEc1 = { consulPort ->
+                EnvoyControlRunnerTestApp(properties = properties(sourceIpClient.ipAddress()), consulPort = consulPort) },
+                envoys = 2,
+                envoyConfig = echoConfig,
+                secondEnvoyConfig = echo2Config
+            )
+            echo3Envoy = createEnvoyContainerWithEcho3Certificate(configOverride = echo3Config)
+            echo3Envoy.start()
+
+            registerServiceWithEnvoyOnIngress(name = "echo", envoy = echoEnvoy, tags = listOf("mtls:enabled"))
+            registerServiceWithEnvoyOnIngress(name = "echo2", envoy = echo2Envoy, tags = listOf("mtls:enabled"))
+
+            waitForEnvoysInitialized()
+        }
+
+        private fun waitForEnvoysInitialized() {
+            untilAsserted {
+                assertThat(echo3Envoy.admin().isEndpointHealthy("echo", echoEnvoy.ipAddress())).isTrue()
+                assertThat(echo3Envoy.admin().isEndpointHealthy("echo2", echo2Envoy.ipAddress())).isTrue()
+                assertThat(echoEnvoy.admin().isEndpointHealthy("echo2", echo2Envoy.ipAddress())).isTrue()
+                assertThat(echo2Envoy.admin().isEndpointHealthy("echo", echoEnvoy.ipAddress())).isTrue()
+            }
+        }
+    }
+
+    fun `should allow echo3 to access status endpoint over https`() {
+        // when
+        val echoResponse = call(service = "echo", from = echo3Envoy, path = "/status/hc")
+
+        // then
+        assertThat(echoResponse).isOk().isFrom(echoLocalService)
+        assertThat(echoEnvoy.ingressSslRequests).isOne()
+        assertThat(echoEnvoy).hasNoRBACDenials()
+
+        // when
+        val echo2Response = call(service = "echo2", from = echo3Envoy, path = "/status/hc")
+
+        // then
+        assertThat(echo2Response).isOk().isFrom(echo2LocalService)
+        assertThat(echo2Envoy.ingressSslRequests).isOne()
+        assertThat(echo2Envoy).hasNoRBACDenials()
+    }
+
+    @Test
+    fun `echo should allow access to status endpoint by all clients over http`() {
```

Comment:
echo is the one with `blockAndLog` and echo2 has `log` this test is the same as the one below and does not tell me anything why it is important. I would go with `should allow access to status endpoint by all clients over http with blockAndLog unlistedEndpointsPolicy`

Reply:
These tests just methodologically iterate over all clients and all servers and test if given client has (or has not) access to given server. The reason why is in proxy settings listed in the test setup. 
Do you prefer to write the reason explicitly in the names of every of these tests? Because it does not apply just to this test, but to every of them.